core/src/main/java/org/elasticsearch/index/mapper/SourceToParse.java

```diff
@@ -91,6 +97,10 @@ public String routing() {
         return this.routing;
     }
 
+    public XContentType getXContentType() {
+        return this.xContentType;
```

Comment:
If this ends up being bigger that it seems, open an issue and we can fix it later. Time to get this PR in.